### DISCUSSION TITLE
I have a solis hybrid inverter with a wifi datalogger with a sn which includes an 'A'

### DISCUSSION THREAD

**Discussion by hornet877** [StephanJoubert/home_assistant_solarman#257] [Q&A]
is there a chance of an update to include wifi datalogger connection :)

**Comment by darkranger008** [comment]
I have the same question.
My Solis-4G Inverter has the Serial Number: 160E31196030082 connected to an old LAN-Datalogger (DLS-L)

why cant i insert the "e" in the serial number? 
Thanks for looking into it..

**Reply by davidrapan** [reply]
That's interesting. Are you sure that it's the serial number of the stick logger and not of the device? Can you show me picture of stick logger's website?

**Comment by darkranger008** [comment]
I am so sorry that I had 2 big Mistakes in my Question.
1st: its a Solis Mini-4G.
2nd: I am so so stupid. I had read the wrong Serial number.
but i cant connect to the DLS-L! no matter which integration i use.


i had then buyed a new S2-WL-ST (S/N: 7A123B1609E00D6D)  <-- this one has Letters in it!


![IMG_5989](https://github.com/user-attachments/assets/a84ca482-630a-4617-aed0-8807258a8949)

**Reply by davidrapan** [reply]
So as I thought this is serial number of the device but you need to use serial number of the solarman stick logger. That one does not contain letters. 😉

**Reply by davidrapan** [reply]
I suggest you to try my integration: https://github.com/davidrapan/ha-solarman

There during the configuration it will try to find your device on the network and autofill said serial number if is the discovery successful.

Edit: Before and if you decide to try my integration. Remove Stephans first from the HACS store as it will prevent potential problems in the future for you. 😉

**Comment by darkranger008** [comment]
Hallo.
thanks for ypur Help. but nothing is working. 
Did not find the device. Tried the both connections LAN and WLAN!

I think it is a problem with the inverter itself!
(Soliscloud is working without problems)

but i never see data of the Inverter in the WEBView of the Datalogger
![device_wlan](https://github.com/user-attachments/assets/59269103-9a78-4aeb-a7c4-8957c472add1)
![device_LAN](https://github.com/user-attachments/assets/bbe451a1-3ef0-4519-af88-8a2edd4bed26)

**Reply by davidrapan** [reply]
DId you also try my integration?

**Comment by darkranger008** [comment]
yes. didnt find anything on the Network itself. and, as you know,  i cant insert the Logger Serial . :)

I also had modified my "old" DLS-L datalogger with RS485 output. I see data coming on modbus(COM/RTU) through a program (qmodmaster). but never get it into HomeAssistant!

**Reply by davidrapan** [reply]
Can you please try that again with debug logging enabled and post the log file here? I'm really interested in this behavior! I think we can sort it out!

Before you click on the Add button enable this:
![350130101-589ced82-4953-48d8-8ea3-683dff670836](https://github.com/user-attachments/assets/340d11af-d930-4653-a6bb-bef5e8401a9b)

After you are done click on that again and it will download file. Post the file here.

**Comment by darkranger008** [comment]
here please:
but i could test more if i can insert "letters" into serial!

[home-assistant_solarman_2024-08-15T17-28-22.338Z.log](https://github.com/user-attachments/files/16628051/home-assistant_solarman_2024-08-15T17-28-22.338Z.log)

**Reply by davidrapan** [reply]
Hm so no response from the device, sad.

Hm you know I look into it and what removing the constraint to only numbers would mean and I'll get back to you! 😉

**Comment by darkranger008** [comment]
hello, anything new about that?

**Reply by davidrapan** [reply]
Hey, After a bit of research it looks like we are out of luck. :-/ I found some info where are ppl describing similar device to yours (serial number with letters) https://github.com/jmccrohan/pysolarmanv5/issues/8

**Comment by darkranger008** [comment]
they talk about the "S3-WLAN-ST" mine is a "S2-WL-ST" with firmware : 10010119

looks like the "S2-WL-ST" is a mixture of all..

https://community.home-assistant.io/t/getting-data-from-solis-inverter/302189/116?page=7

https://github.com/fboundy/ha_solis_overview

**Reply by davidrapan** [reply]
I know that, but it's obvious that both devices are sharing some common hardware.

**Comment by darkranger008** [comment]
Shit, thats bad for me... :)
But anyway, thanks for your Help!

**Reply by davidrapan** [reply]
Yeah well...

Sure thing! Np.. 😉
